### PR TITLE
[PLA-1840] add account update for sign

### PR DESCRIPTION
**Pull request by zlayine**
### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Updated `SignTransaction.vue` to include logic for updating user wallet account if the application is multi-tenant.
- Added `updateUserAccount` method to `AuthApi` class in `auth.ts`.
- Modified `UpdateUser` GraphQL mutation to accept an `account` variable.
- Added `updateUserAccount` method to `useAppStore` in `index.ts`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>SignTransaction.vue</strong><dd><code>Update user wallet account in multi-tenant setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/components/SignTransaction.vue

<li>Imported <code>publicKeyToAddress</code> from <code>~/util/address</code>.<br> <li> Added logic to update user wallet account if multi-tenant.<br>

</details>

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/131/files#diff-8beca3bdc6075e1ec224d0a348723ac92bdf4cf83a8285008a4cd0431945652c">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>auth.ts</strong><dd><code>Add updateUserAccount method in AuthApi</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/api/auth.ts

- Added `updateUserAccount` method to `AuthApi` class.

</details>

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/131/files#diff-fd04f32c9611745605e49b95c6579fe737ef7ccd849966fa82093e095af5f0bc">+11/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UpdateUser.ts</strong><dd><code>Extend UpdateUser mutation to include account variable</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/graphql/mutation/auth/UpdateUser.ts

- Modified `UpdateUser` mutation to include `account` variable.

</details>

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/131/files#diff-2a28d3e2d257f27d0a535e2f485879e4b88a1b49b1c59027a04add8ad1688be5">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Add updateUserAccount method in useAppStore</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/store/index.ts

- Added `updateUserAccount` method to `useAppStore`.

</details>

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/131/files#diff-45cbdaed3eae4593b455daabf8cbc593b4d1f00058757cf6ca25a1b9967ebda2">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: resources/js/graphql/mutation/auth/UpdateUser.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
-export default `mutation UpdateUser($email: String, $password: String) {
-    UpdateUser(email: $email, password: $password)
+export default `mutation UpdateUser($email: String, $password: String, $account: String) {
+    UpdateUser(email: $email, password: $password, account: $account)
```
Comment (enjinabner) [inline resources/js/graphql/mutation/auth/UpdateUser.ts:2]: This attribute is dedicated for wallet daemon and we cannot update this. I created a new attribute for this in this PR https://github.com/enjin/platform-multi-tenant/pull/72

**Review by enjinabner — CHANGES_REQUESTED** [review]
updates

File: resources/js/graphql/mutation/auth/UpdateUser.ts (hunk 1 above)
Comment (zlayine) [reply resources/js/graphql/mutation/auth/UpdateUser.ts:2]: okay thank you! will check the new implementation

**Comment by zlayine** [comment]
update made with another ticket #135

**Pull request closed without merging**